src/core/instance/state.js

```diff
@@ -67,8 +68,9 @@ function initProps (vm: Component, props: Object) {
         }
       })
     } else {
-      defineReactive(vm, key, validateProp(key, props, propsData, vm))
+      defineReactive(vm.$props, key, validateProp(key, props, propsData, vm))
     }
+    proxy(vm, '$props', key)
```

Comment:
There is a problem here.
I think that is... if you want to do this.
You need to create a proxy on vm, not on `$props`
Or may bring destructive changes